kadi/commands/core.py

```diff
@@ -132,25 +139,129 @@ def vstack_exact(tables):
     return out
 
 
-def read_backstop(backstop):
+def read_backstop(
+    backstop: str | Path,
+    add_observations: bool = False,
+    load_name=None,
+) -> "CommandTable":
     """Read ``backstop`` and return a ``CommandTable``.
 
-    The ``backstop`` argument can be either be a string file name or a backstop
-    table from ``parse_cm.read_backstop``.
+    The ``backstop`` argument can be either be a string file name or path.
 
-    This function is a wrapper around ``get_cmds_from_backstop`` but follows a
-    more typical naming convention.
+    If ``add_observations`` is ``True`` then add OBS commands (default=False) into the
+    returned ``CommandTable``. This allows the returned ``CommandTable`` to be used in
+    calls to ``get_observations`` and ``get_starcats``.
+
+    If ``load_name`` is provided then this is used for the "source" column in the
+    returned table. If not provided, the default is to use the load name if it can be
+    determined from the backstop directory path (e.g. "DEC1123A" from
+    "<rootdir>/2023/DEC1123/oflsa/CR344_2303.backstop"), otherwise it is set to "None".
```

Comment:
Maybe the regex as well as an example, as the first thing I did was read a backstop from a local 'DEC1123A' directory and then get momentarily stumped about why it was not marked as load_name/source DEC1123A.

Reply:
Done, I hope, in [7f875f1](https://github.com/sot/kadi/pull/297/commits/7f875f1d0affe93d46247fd4792124e0a039688a)

Reply:
Thanks!